pkg/controller/flink/flink.go

```diff
@@ -140,6 +140,14 @@ func getURLFromApp(application *v1beta1.FlinkApplication, hash string) string {
 	return fmt.Sprintf("http://%s.%s:%d", service, application.Namespace, port)
 }
 
+func getGenericURLFromApp(application *v1beta1.FlinkApplication) string {
```

Comment:
This seems to be the `service` url, and is not exposed outside of the kubernetes cluster. Is that the intention ?

Reply:
Hmm, this seems to be exposed in the local environment. i.e. for example for a locally running app — http://localhost:8001/api/v1/namespaces/default/services/simpleflinkk8s:8081/proxy/#/overview is exposed. But maybe works differently in a non-local environment?

Reply:
Yeah, this should be returning the external URL, which for non-proxy setups is defined by the ingress.

E.g. for our clusters it would look like, `<appname>.ingress.core-staging-4.us-east-1.k8s.lyft.net`

Reply:
Okay makes sense. I updated this for non-proxy setups to return `GetFlinkUIIngressURL`.